src/lib/core/components/filter/HistogramFilter.tsx

```diff
@@ -769,9 +770,15 @@ function formatStatValue(
 ) {
   return type === 'date'
     ? String(value).replace(/T.*$/, '')
-    : Number(value).toLocaleString(undefined, {
-        maximumFractionDigits: 4,
-      });
+    : // need to check integer not to add additional decimal points
+    Number(value) % 1 === 0
+    ? changeNumberWithCommas(Number(value))
+    : changeNumberWithCommas(Number(value).toFixed(4));
+}
+
+// a function to add comma for a number > 9999
+function changeNumberWithCommas(x: number | string) {
```

Comment:
If the true purpose is to avoid years being rendered as 2,021, it might be better to continue to use `toLocaleString()` with the `useGrouping` option set to false for numbers in the range (1900-2100) ?

Reply:
@bobular thank you for your suggestion 👍 Will think about this